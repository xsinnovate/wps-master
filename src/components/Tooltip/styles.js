import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  overlay: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 20,
    backgroundColor: 'rgba(0,0,0,0.5)',
  },
  modal: {
    paddingHorizontal: 20,
    paddingVertical: 25,
    backgroundColor: '#fff',
  },
});
