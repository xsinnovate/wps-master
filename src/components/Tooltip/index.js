import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import {
  Modal,
  TouchableHighlight,
  TouchableWithoutFeedback,
  Text,
  View,
} from 'react-native';
import styles from './styles';

export default class Tooltip extends PureComponent {
  state = {
    visible: false,
  };

  onToggle = () => {
    this.setState(state => ({ visible: !state.visible }));
  }

  render() {
    const { children, text } = this.props;
    const { visible } = this.state;

    return (
      <>
        <TouchableHighlight
          activeOpacity={0.5}
          onPress={this.onToggle}
          underlayColor="#fff"
        >
          {children}
        </TouchableHighlight>
        <Modal
          animationType="fade"
          transparent
          visible={visible}
        >
          <TouchableWithoutFeedback onPress={this.onToggle}>
            <View style={styles.overlay}>
              <View style={styles.modal}>
                <Text>
                  {text}
                </Text>
              </View>
            </View>
          </TouchableWithoutFeedback>
        </Modal>
      </>
    );
  }
}

Tooltip.propTypes = {
  children: PropTypes.node.isRequired,
  text: PropTypes.string.isRequired,
};
