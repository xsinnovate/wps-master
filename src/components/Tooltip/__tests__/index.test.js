import React from 'react';
import renderer from 'react-test-renderer';
import {
  Modal,
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback,
} from 'react-native';
import Tooltip from '..';

const content = <Text>Text</Text>;

describe('<Tooltip />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Tooltip text="Tip">{content}</Tooltip>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should display modal when pressed', () => {
    const instance = renderer.create(<Tooltip text="Tip">{content}</Tooltip>).root;

    instance.findByType(TouchableHighlight).props.onPress();

    expect(instance.findByType(Modal).props.visible).toBe(true);
  });

  it('should hide modal when overlay pressed', () => {
    const instance = renderer.create(<Tooltip text="Tip">{content}</Tooltip>).root;

    instance.findByType(TouchableHighlight).props.onPress();
    instance.findByType(TouchableWithoutFeedback).props.onPress();

    expect(instance.findByType(Modal).props.visible).toBe(false);
  });
});
