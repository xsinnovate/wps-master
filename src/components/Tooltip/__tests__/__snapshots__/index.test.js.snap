// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> renders correctly 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={null}
    >
      Text
    </Text>
  </View>,
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "paddingHorizontal": 20,
            "paddingVertical": 25,
          }
        }
      >
        <Text>
          Tip
        </Text>
      </View>
    </View>
  </Modal>,
]
`;
